charts/ratify/templates/assign.yaml

```diff
@@ -0,0 +1,89 @@
+{{- if semverCompare ">= 3.11.0" .Values.gatekeeper.version }}
+apiVersion: mutations.gatekeeper.sh/v1
+{{- else }}
+apiVersion: mutations.gatekeeper.sh/v1beta1
+{{- end }}
+kind: Assign
+metadata:
+  name: mutate-pod-image
+spec:
+  match:
+    scope: Namespaced
+    kinds:
+    - apiGroups: ["*"]
+      kinds: ["Pod"]
+    excludedNamespaces:
+      - "gatekeeper-system"
+      {{- if ne .Release.Namespace "gatekeeper-system"}}
+      - {{ .Release.Namespace }}
+      {{- end }}
+  applyTo:
+  - groups: [""]
+    kinds: ["Pod"]
+    versions: ["v1"]
+  location: "spec.containers[name:*].image"
+  parameters:
+    assign:
+      externalData:
+        provider: ratify-mutation-provider
+
+---
+{{- if semverCompare ">= 3.11.0" .Values.gatekeeper.version }}
+apiVersion: mutations.gatekeeper.sh/v1
+{{- else }}
+apiVersion: mutations.gatekeeper.sh/v1beta1
+{{- end }}
```

Comment:
Make this a named template, move it to `_helpers.tpl` (xref: https://helm.sh/docs/chart_template_guide/named_templates/) and reuse in this template.